src/platform.ts

```diff
@@ -223,9 +229,39 @@ export class PowerShellExeFinder {
     private *enumerateAdditionalPowerShellInstallations(): Iterable<IPossiblePowerShellExe> {
         for (const versionName in this.additionalPowerShellExes) {
             if (Object.prototype.hasOwnProperty.call(this.additionalPowerShellExes, versionName)) {
-                const exePath = this.additionalPowerShellExes[versionName];
-                if (exePath) {
-                    yield new PossiblePowerShellExe(exePath, versionName);
+                let exePath = this.additionalPowerShellExes[versionName];
+                if (!exePath) {
+                    continue;
+                }
+
+                // Remove surrounding quotes from path (without regex)
+                if (exePath.startsWith("'") && exePath.endsWith("'")
+                    || exePath.startsWith("\"") && exePath.endsWith("\"")) {
+                    exePath = exePath.slice(1, -1);
+                }
+
+                // Always search for what the user gave us first
+                yield new PossiblePowerShellExe(exePath, versionName);
+
+                // Also search for `pwsh[.exe]` and `powershell[.exe]` if missing
+                const args: [string, undefined, boolean, boolean]
+                    // Must be a tuple type and is suppressing the warning
+                    = [versionName, undefined, true, true];
+
+                // Handle Windows where '.exe' and 'powershell' are things
+                if (this.platformDetails.operatingSystem === OperatingSystem.Windows) {
+                    if (!exePath.endsWith("pwsh.exe") && !exePath.endsWith("powershell.exe")) {
```

Comment:
Agree probably need some tests to validate all these potential permutations, looks good to me though.

Reply:
I realized last night I should also be able to handle the common case of `~/`, and share that and the quote-stripping with the `cwd` setting, and then do a preview and _then_ work on saving the `cwd` in a cache instead of a setting.

Reply:
Ok so I decided that since the Python extension uses `untildify` I'm fine bringing it too, which means I'll get that taken care of...in the next PR. @JustinGrote could you give this a review? It's got tests now!